### ISSUE TITLE
Should shuffle data while training

### ISSUE THREAD

**Issue by KaidongYu** [IBM/pytorch-seq2seq#96]
Very Nice Seq2Seq Implement :)

I'm using this module for Text Generation, I found that the training data order is exact same for every epoch. The source code is:
```
batch_iterator = torchtext.data.BucketIterator(
            dataset=data, batch_size=self.batch_size,
            sort=True, sort_key=lambda x: len(x.src),
            device=device, repeat=False)
```

Maybe we should set `sort=False` to shuffle the data and `sort_within_batch=True` to sort the lengths array in decreasing order

**Comment by kylegao91** [comment]
Yes it makes sense.  Would you like to make a pr about it?

**Comment by KaidongYu** [comment]
@kylegao91 Sure, I pushed the PR on #97
